src/Microsoft.VisualStudio.Threading.Analyzers/CommonInterest.cs

```diff
@@ -63,9 +63,11 @@ internal static class CommonInterest
 
         private const RegexOptions FileNamePatternRegexOptions = RegexOptions.IgnoreCase | RegexOptions.Singleline;
 
-        private static readonly Regex NegatableTypeOrMemberReferenceRegex = new Regex(@"^(?<negated>!)?\[(?<typeName>[^\[\]\:]+)+\](?:\:\:(?<memberName>\S+))?\s*$", RegexOptions.Singleline | RegexOptions.CultureInvariant);
+        private static readonly TimeSpan RegexMatchTimeout = TimeSpan.FromMilliseconds(100);  // Prevent expensive CPU hang in Regex.Match if backtracking occurs due to pathological input (see #485).
```

Comment:
Timeouts this small seem very likely to misfire. We have "fast" tests with 3 second timeouts that still timeout occasionally, driving me to use nothing less than 5 second timeouts for anything that would report a failure rather than retry. Computers can just be slow sometimes.
```suggestion
        private static readonly TimeSpan RegexMatchTimeout = TimeSpan.FromSeconds(5);  // Prevent expensive CPU hang in Regex.Match if backtracking occurs due to pathological input (see #485).
```